roles/network_plugin/flannel/templates/cni-flannel.yml.j2

```diff
@@ -66,7 +66,7 @@ spec:
       serviceAccountName: flannel
       containers:
       - name: kube-flannel
-        image: {{ flannel_image_repo }}:{{ flannel_image_tag }}-{{ arch }}
+        image: {{ flannel_image_repo }}:{{ flannel_image_tag | regex_replace(image_arch,'') }}-{{ arch }}
```

Comment:
I was wondering why we need to remove and add similar string here, but that seems the same as the existing code https://github.com/kubernetes-sigs/kubespray/blob/10083fa47e579b49f7cd9736556bd9390e1ecb92/roles/network_plugin/flannel/templates/cni-flannel.yml.j2#L69

So it is fine to do that.